Cesium.Parser/CParser.cs

```diff
@@ -271,7 +271,7 @@ public partial class CParser
         IDirectDeclarator @base,
         ICToken _,
         IdentifierList? identifierList,
-        ICToken __) => new IdentifierListDirectDeclarator(@base, identifierList);
+        ICToken __) => new IdentifierListDirectDeclarator(@base, identifierList ?? IdentifierList.Empty);
```

Comment:
The general rule I'm trying to follow in the main AST is that if something's defined as optional by the standard, then it should be `null` in the AST.

Otherwise, corner cases such as empty function parameter list vs a void one become much harder to track.

Intermediate AST (aka "IR") may be less restrictive, but regarding this particular change I'm not sure.